packages/grafana-ui/src/components/Table/Table.tsx

```diff
@@ -179,26 +187,41 @@ export const Table: FC<Props> = memo((props: Props) => {
       data: memoizedData,
       disableResizing: !resizable,
       stateReducer: stateReducer,
-      initialState: getInitialState(initialSortBy, memoizedColumns),
+      initialState: getInitialState(initialSortBy, memoizedColumns, pageSize),
       sortTypes: {
         number: sortNumber, // the builtin number type on react-table does not handle NaN values
         'alphanumeric-insensitive': sortCaseInsensitive, // should be replace with the builtin string when react-table is upgraded, see https://github.com/tannerlinsley/react-table/pull/3235
       },
     }),
-    [initialSortBy, memoizedColumns, memoizedData, resizable, stateReducer]
+    [initialSortBy, memoizedColumns, memoizedData, pageSize, resizable, stateReducer]
   );
 
-  const { getTableProps, headerGroups, rows, prepareRow, totalColumnsWidth, footerGroups } = useTable(
-    options,
-    useFilters,
-    useSortBy,
-    useAbsoluteLayout,
-    useResizeColumns
-  );
+  const {
+    getTableProps,
+    headerGroups,
+    rows,
+    prepareRow,
+    totalColumnsWidth,
+    footerGroups,
+    page,
+    state,
+    setPageSize,
+    gotoPage,
+    pageOptions,
+  } = useTable(options, useFilters, useSortBy, usePagination, useAbsoluteLayout, useResizeColumns);
+
+  useEffect(() => {
+    if (pageSize) {
+      setPageSize(pageSize);
+    }
+  }, [pageSize, setPageSize]);
```

Comment:
this will always make it so that panels have two initial renders, is there a way to only call setPageSize when it has changed from initial state?

Reply:
@torkelo Can you help me where you see that? I've added a console log and I only see this run once also checked the redux devtools and the didn't see the initial render twice.

Reply:
the useEffect will run once, and trigger a re-render , setPageSize will always be called on mount to set the same initial page size the table was initialized with